src/crate/client/sqlalchemy/__init__.py

```diff
@@ -19,7 +19,13 @@
 # with Crate these terms will supersede the license and you may use the
 # software solely pursuant to the terms of the relevant commercial agreement.
 
+from .compat.api13 import monkeypatch_add_exec_driver_sql
 from .dialect import CrateDialect
+from .sa_version import SA_1_4, SA_VERSION
+
+# SQLAlchemy 1.3 does not have the `exec_driver_sql` method.
+if SA_VERSION < SA_1_4:
+    monkeypatch_add_exec_driver_sql()
```

Comment:
Do we already know how much longer we intend to support SQLAlchemy 1.3? According to SQLAlchemy's [version policy](https://www.sqlalchemy.org/download.html#status-desc-maintenance), 1.3 will become EOL as soon as 2.0 becomes stable. Given that maintaining compatibility with 1.3 here reaches a certain complexity, do we want to keep it much longer?

Reply:
Hi. The most recent discussion about EOL of SA13 can be found at https://github.com/sqlalchemy/sqlalchemy/issues/8631#issuecomment-1329264864. I think we will support it for a while, and I am happy that the changes have not been too much intrusive, other than needing a few lines in `compat/api13.py`.

We discussed different alternatives on maintenance of the code within this package with @seut, regarding version deprecations and such, but all other alternatives would need more efforts, especially if we would need to deprecate support for SA13 earlier, because this might mean we would have to keep a maintenance branch or such. See also GH-494.

Reply:
Nevertheless, I think it will not hurt to add a runtime warning for the next release, which will give users a notice that dropping support for SA13 will be around the corner.

Reply:
The 1.3 support is quite nicely encapsulated, indeed. Well done! For now (while 1.3 is still supported upstream) it's a good solution. Once 1.3 reached EOL, I would still vote for removing it shortly after to eliminate the need for monkey patching (which always feels a bit risky to me, especially in tests).

Reply:
+1 for removing 1.3 support. Given that crate-python is mostly stable (bugfixes are rare) we can point users of SQLAlchemy 1.3 to an older version of crate-python and make sure to update the version constraints for the next crate-python release. `pip` should then pick the right version, no?

Reply:
I agree. See also GH-501.